right-for-life_front/src/components/FloatingBtn/BackBtn.jsx

```diff
@@ -1,12 +1,17 @@
 import React from "react";
 import { FloatingBtn } from "./FloatingBtn.jsx";
+import { connect } from 'react-redux';
 
-export const BackBtn = ({position}) => (
+const BackBtn = ({position, useOrNotGoBack}) => (
   <div className="relative width-full">
     <FloatingBtn
       icon="fa-arrow-left"
       position={position}
-      onClick={window.history.length === 1 ? null : () => window.history.back()}
-      url="/"/>
+      onClick={useOrNotGoBack ? () => window.history.forward() || window.history.back() : null}
+      url={window.location.pathname.match(/\/[a-z]+/)[0]}/>
   </div>
 );
+
+const withConnect = connect(state => ({useOrNotGoBack: state.useOrNotGoBack}))(BackBtn);
```

Comment:
[optional] If you use a functional component, you can use "useSelector" hook from Redux, instead of the higher-order component "connect".

Reply:
Thanks for the review! I added click tracking of the Back and Forward buttons, so the component must be subscribed to the store. So while there is no way not to use Connect. But in the future I think I will use hooks instead of decorators, thanks